### ISSUE TITLE
Supported input type 'file'

### ISSUE THREAD

**Issue by shajra** [edolstra/flake-compat#53]
I have a flake that downloads an EXE and unpacks it to get to an Electron app that's inside it for unofficially supported Linux support.  I've previously done something similar with MacOS DMG files.

I think there will always be pointers to various files on the internet that we don't have controlled fetchers for.  Flakes seems to be able to handle these pretty well with an input type of `file`.  Is there any reason this can't be supported with `flake-compat`?

I think a solution could give this project what `getFlakes` returns for these `file` inputs:

```nix
{
    narHash = "sha256-…=";
    outPath = "/nix/store/…-source";
}
```

**Comment by grantshandy** [comment]
I'm getting this issue too, downloading an arbitrary JS file. Adding file input support would fix my package. Thanks!

**Comment by edolstra** [comment]
Yes, this is what the `file` input type is for. But it's currently unsupported by `flake-compat` (see #44).
